tactical-microgrid-standard/common/TimerHandler.h

```diff
@@ -0,0 +1,246 @@
+#ifndef TMS_TIMER_HANDLER_H
+#define TMS_TIMER_HANDLER_H
+
+#include <ace/Event_Handler.h>
+#include <ace/Log_Msg.h>
+#include <ace/Reactor.h>
+
+#include <chrono>
+#include <map>
+#include <variant>
+#include <mutex>
+#include <type_traits>
+#include <memory>
+#include <stdexcept>
+
+using Sec = std::chrono::seconds;
+using Clock = std::chrono::system_clock;
+using TimePoint = std::chrono::time_point<Clock>;
+using TimerId = long;
+constexpr TimerId null_timer_id = 0;
+using Mutex = std::recursive_mutex;
```

Comment:
Can this use `std::mutex` instead?

Reply:
Maybe, but it would require changing a lot of how the locking is down.